cookbooks/aws-parallelcluster-environment/recipes/init/config_default_user_home.rb

```diff
@@ -54,8 +59,30 @@
     rsync -a /tmp#{node['cluster']['cluster_user_home']}/ #{node['cluster']['cluster_user_local_home']}
     usermod -d #{node['cluster']['cluster_user_local_home']} #{node['cluster']['cluster_user']}
     chown -R #{node['cluster']['cluster_user']}: #{node['cluster']['cluster_user_local_home']}
-    rm -rf /tmp#{node['cluster']['cluster_user_home']}
-    rm -rf #{node['cluster']['cluster_user_home']}
+  EOH
+end
+
+# Data integrity check
+# 1. Verifies data integrity by comparing the temporary backup directory and the new local home directory.
+# 2. If the data integrity check passes, it removes both the temporary backup directory and the original home directory.
+# 3. If the data integrity check fails, it outputs an error message and exits with an error code 1.
+# 4. Unhappy Path Manually test passed successfully:
```

Comment:
No need to specify the tests you made in a comment. It's enough to describe it in the PR description.

Reply:
Done!